flink-ml-lib/src/test/java/org/apache/flink/ml/feature/MaxAbsScalerTest.java

```diff
@@ -0,0 +1,251 @@
+
+        trainSparseDataTable =
+                tEnv.fromDataStream(env.fromCollection(TRAIN_SPARSE_DATA)).as("input");
+        predictSparseDataTable =
+                tEnv.fromDataStream(env.fromCollection(PREDICT_SPARSE_DATA)).as("input");
+    }
+
+    private static void verifyPredictionResult(
+            Table output, String outputCol, List<Vector> expectedData) throws Exception {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) output).getTableEnvironment();
+
+        DataStream<Vector> stream =
+                tEnv.toDataStream(output)
+                        .map((MapFunction<Row, Vector>) row -> row.getFieldAs(outputCol));
+
+        List<Vector> result = IteratorUtils.toList(stream.executeAndCollect());
+        compareResultCollections(expectedData, result, TestUtils::compare);
+    }
+
+    @Test
+    public void testParam() {
+        MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+        assertEquals("input", maxAbsScaler.getInputCol());
+        assertEquals("output", maxAbsScaler.getOutputCol());
+
+        maxAbsScaler.setInputCol("test_input").setOutputCol("test_output");
+        assertEquals("test_input", maxAbsScaler.getInputCol());
+        assertEquals("test_output", maxAbsScaler.getOutputCol());
+    }
+
+    @Test
+    public void testOutputSchema() {
+        MaxAbsScaler maxAbsScaler =
+                new MaxAbsScaler().setInputCol("test_input").setOutputCol("test_output");
+
+        MaxAbsScalerModel model = maxAbsScaler.fit(trainDataTable.as("test_input"));
+        Table output = model.transform(predictDataTable.as("test_input"))[0];
+        assertEquals(
+                Arrays.asList("test_input", "test_output"),
+                output.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testFitAndPredict() throws Exception {
+        MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+        MaxAbsScalerModel maxAbsScalerModel = maxAbsScaler.fit(trainDataTable);
+        Table output = maxAbsScalerModel.transform(predictDataTable)[0];
+        verifyPredictionResult(output, maxAbsScaler.getOutputCol(), EXPECTED_DATA);
+    }
+
+    @Test
+    public void testFitDataWithNullValue() {
+        List<Row> trainData =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Row.of(Vectors.dense(0.0, 3.0)),
+                                Row.of(Vectors.dense(2.1, 0.0)),
+                                Row.of((Object) null),
+                                Row.of(Vectors.dense(6.1, 8.1)),
+                                Row.of(Vectors.dense(200, 400))));
+
+        Table trainDataWithInvalidData =
+                tEnv.fromDataStream(env.fromCollection(trainData)).as("input");
+
+        try {
+            MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+            maxAbsScaler.fit(trainDataWithInvalidData);
```

Comment:
Could you insert a `fail();` after this line to verify that the code never goes there?